lib/beaker-puppet/helpers/facter_helpers.rb

```diff
@@ -44,7 +44,7 @@ def fact_on(host, name, opts = {})
             opts << ' --json'
           end
 
-          result = on host, facter(name, opts)
+          result = on host, facter("\"#{name}\"", opts)
```

Comment:
An example of a fact name with spaces is https://github.com/ekinanp/facter/commit/31d34005c14d9c33ed1ab42b0d1810f5b9412df0#diff-1796fc5451255bbafbb615c5ce6ed0faR29 where on Windows, `primary_interface` has a space in it.